src/components/Frame/Frame.scss

```diff
@@ -126,10 +134,20 @@ $skip-vertical-offset: rem(10px);
   width: 100%;
   height: top-bar-height();
 
+  @include frame-when-nav-displayed {
+    left: var(--p-frame-offset);
+    width: calc(100% - var(--p-frame-offset));
+  }
+
   &.TopBar-newDesignLanguage {
     @include frame-when-nav-displayed {
-      left: layout-width(unstable-nav);
-      width: calc(100% - #{layout-width(unstable-nav)});
+      // stylelint-disable-next-line selector-max-class
+      .hasNav & {
+        left: calc(#{layout-width(unstable-nav)} + var(--p-frame-offset));
```

Comment:
Can you update this `unstable-nav` name in another PR?

Reply:
https://github.com/Shopify/polaris-react/pull/2890